lib/cdo/high_frequency_reporter.rb

```diff
@@ -0,0 +1,76 @@
+require 'csv'
+require 'time'
+# Usage:
+#   errors = HighFrequencyReporter.new(Slack.new)
+#   errors.load!
+#   loop do
+#     <stuff>
+#     errors.record "Something went wrong"
+#   end
+#   errors.report!
+#   errors.save
+class HighFrequencyReporter
+  attr_accessor :new_events
+
+  def initialize(chat_client, channel, log_file_name)
+    @chat_client = chat_client
+    @channel = channel
+    # Load old events from disk
+    # [{name: "eventblah", reported_at: '2020010...'}, ...]
+    @old_events = []
+    @new_events = []
+    @filename = log_file_name
+  end
+
+  # Loads known events from previous runs from a file on disk
+  def load
+    # Format for log file: CSV
+    # @see https://ruby-doc.org/stdlib-2.6.1/libdoc/csv/rdoc/CSV.html
+    # Each row:
+    #   "error message","timestamp"
```

Comment:
I assume the header for timestamp column is `reported_at`? Could you add 1 line showing all column headers here.

Usually log file has timestamp first because it has fixed length (while event message length varies) and it helps quickly show time difference among events.